Library/Homebrew/cmd/upgrade.rb

```diff
@@ -18,10 +18,15 @@ def upgrade
     Homebrew.perform_preinstall_checks
 
     if ARGV.named.empty?
-      outdated = Formula.installed.select(&:outdated?)
+      outdated = Formula.installed.select do |f|
+        f.outdated?(:fetch_head => ARGV.fetch_head?)
+      end
+
       exit 0 if outdated.empty?
     elsif ARGV.named.any?
-      outdated = ARGV.resolved_formulae.select(&:outdated?)
+      outdated = ARGV.resolved_formulae.select do |f|
+        f.outdated?(:fetch_head => ARGV.fetch_head?)
+      end
```

Comment:
Any ideas on how we can achieve it? I can think of adding some methods to `ARGV` for that purpose, but that's not the best solution I guess.